bezier-rs/lib/src/lib.rs

```diff
@@ -453,15 +460,89 @@ impl Bezier {
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let transformed_handle_start = transformation_function(handle_start);
+				let transformed_handle_end = transformation_function(handle_end);
+				Bezier::from_cubic_dvec2(transformed_start, transformed_handle_start, transformed_handle_end, transformed_end)
+			}
+		}
+	}
+
+	/// Returns a Bezier curve that results from rotating the curve by the given angle (in radians).
+	pub fn rotate(&self, angle: f64) -> Bezier {
+		let rotation_matrix = DMat2::from_angle(angle);
+		self.apply_transformation(&|point| rotation_matrix.mul_vec2(point))
+	}
+
+	/// Returns a Bezier curve that results from translating the curve by the given `DVec2`.
+	pub fn translate(&self, translation: DVec2) -> Bezier {
+		self.apply_transformation(&|point| point + translation)
+	}
+
+	/// Returns a list of points where the provided `line` intersects with the Bezier curve.
+	/// - `line` - Expected to be received in the format of `[start_point, end_point]`.
+	pub fn line_intersection(&self, line: [DVec2; 2]) -> Vec<DVec2> {
+		// Rotate the bezier and the line by the angle that the line makes with the x axis
+		let slope = line[1] - line[0];
+		let angle = slope.angle_between(DVec2::new(1., 0.));
+		let rotation_matrix = DMat2::from_angle(angle);
+		let rotated_bezier = self.apply_transformation(&|point| rotation_matrix.mul_vec2(point));
+		let rotated_line = [rotation_matrix.mul_vec2(line[0]), rotation_matrix.mul_vec2(line[1])];
+
+		// Translate the bezier such that the line becomes aligned on top of the x-axis
+		let vertical_distance = rotated_line[0].y;
+		let translated_bezier = rotated_bezier.translate(DVec2::new(0., -vertical_distance));
+
+		// Compute the roots of the resulting bezier curve
+		let list_intersection_t = match translated_bezier.handles {
+			BezierHandles::Quadratic { handle } => {
+				let a = translated_bezier.start.y - 2. * handle.y + translated_bezier.end.y;
+				let b = 2. * (handle.y - translated_bezier.start.y);
+				let c = translated_bezier.start.y;
+				let discriminant = b * b - 4. * a * c;
+				let two_times_a = 2. * a;
+				utils::solve_quadratic(discriminant, two_times_a, b, c)
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let start_y = translated_bezier.start.y;
+				let a = -start_y + 3. * handle_start.y - 3. * handle_end.y + translated_bezier.end.y;
+				let b = 3. * start_y - 6. * handle_start.y + 3. * handle_end.y;
+				let c = -3. * start_y + 3. * handle_start.y;
+				let d = start_y;
+				utils::solve_cubic(a, b, c, d)
+			}
+		};
+		let min = line[0].min(line[1]);
+		let max = line[0].max(line[1]);
+		let max_abs_diff = 1e-4;
+
+		list_intersection_t
+			.iter()
+			.filter(|&&t| utils::f64_approximately_in_range(t, 0., 1., max_abs_diff))
+			.map(|&t| self.unrestricted_compute(t))
+			.filter(|&p| utils::dvec2_approximately_in_range(p, min, max, max_abs_diff).all())
```

Comment:
Does `p` mean `point`?

Reply:
Yup! I'll change it to `point` instead of `p`.